### PR TITLE
Molecularize pulmap

### PR DESCRIPTION
**Pull request by kayiwa**

### CODE REVIEW

File: roles/pulibrary.pulmap/tasks/main.yml [hunk 1]
```diff
@@ -1,42 +1,54 @@
...
     weekday: 3
-    job: "/bin/bash -l -c 'export PATH=\"/usr/local/bin/:$PATH\" && cd /opt/{{rails_app_directory}}/current && RAILS_ENV=production bundle exec rake sitemap:refresh >> /tmp/pulmap_sitemap.log 2>&1'"
+    job: "/bin/bash -l -c 'export PATH=\"/usr/local/bin/:$PATH\" && cd /opt/{{ rails_app_directory }}/current && RAILS_ENV=production bundle exec rake sitemap:refresh >> /tmp/pulmap_sitemap.log 2>&1'"
     minute: 0
-    user: '{{deploy_user}}'
-- name: Create cloud_config directory
-  file:
-    path: '/home/{{deploy_user}}/cloud_config'
-    state: 'directory'
-- name: Install Google Cloud Authorization
```
Comment (eliotjordan) [inline roles/pulibrary.pulmap/tasks/main.yml]: This is needed for thumbnails.

File: roles/pulibrary.pulmap/tasks/main.yml [hunk 2]
```diff
@@ -1,42 +1,54 @@
...
-    owner: '{{deploy_user}}'
-    group: '{{deploy_user}}'
-    state: 'link'
-    force: true
-  with_items:
-    - src: '/mnt/maplab'
-      link: '/opt/{{rails_app_directory}}/shared/maplab'
+    user: "{{ deploy_user }}"
+
+- name: Pulmap | precache thumbnails cron job
```
Comment (eliotjordan) [inline roles/pulibrary.pulmap/tasks/main.yml]: This is no longer needed.

**Comment by eliotjordan** [comment]
@kayiwa Some updates that were made last week have been altered by this PR a bit. I'll take a look and see if I can push up a fix.

File: roles/pulibrary.pulmap/tasks/main.yml (hunk 1 above)
Comment (kayiwa) [reply roles/pulibrary.pulmap/tasks/main.yml]: done!

File: roles/pulibrary.pulmap/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  license: MIT
+
+  min_ansible_version: 2.8
+
+  platforms:
+    - name: Ubuntu
+      versions:
+        - 18.04
+dependencies:
```
Comment (eliotjordan) [inline roles/pulibrary.pulmap/meta/main.yml:14]: Are the `deploy_user`, `passenger`, and `nodejs` roles no longer needed?

File: roles/pulibrary.pulmap/meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  - role: "pulibrary.samba"
+  - role: "pulibrary.ruby"
+  - role: "pulibrary.postgresql"
```
Comment (eliotjordan) [inline roles/pulibrary.pulmap/meta/main.yml:17]: Does this install the postgres server? If so, we don't need that for pulmap.

File: roles/pulibrary.pulmap/meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  - role: "pulibrary.redis"
+  - role: "pulibrary.sidekiq_worker"
+  - role: "pulibrary.sneakers_worker"
+  - role: "pulibrary.blacklight-app"
+  - role: "pulibrary.rails-app"
+  - role: "pulibrary.mailcatcher"
```
Comment (eliotjordan) [inline roles/pulibrary.pulmap/meta/main.yml:23]: This is new. I don't think we need this for Pulmap.

Comment (kayiwa) [reply roles/pulibrary.pulmap/meta/main.yml:23]: do you need to do tests for mail delivery on staging? If yes then you will need it.

File: roles/pulibrary.pulmap/meta/main.yml (hunk 3 above)
Comment (kayiwa) [reply roles/pulibrary.pulmap/meta/main.yml:14]: the rails_app (listed as dependency will bring those in)

File: roles/pulibrary.pulmap/tasks/main.yml [hunk 6]
```diff
@@ -1,42 +1,61 @@
...
-    src: '{{item.src}}'
-    dest: '{{item.link}}'
-    owner: '{{deploy_user}}'
-    group: '{{deploy_user}}'
-    state: 'link'
-    force: true
-  with_items:
-    - src: '/mnt/maplab'
-      link: '/opt/{{rails_app_directory}}/shared/maplab'
+- name: Pulmap| Create a sitemap cron job
```
[the remark this answers is not in the document]
Comment (kayiwa) [reply roles/pulibrary.pulmap/tasks/main.yml:46]: I am assuming a capistrano does this?

**Review by eliotjordan — APPROVED** [review]

**Pull request merged**